cfme/tests/automate/test_service_automate.py

```diff
@@ -139,3 +141,73 @@ def test_user_requester_for_lifecycle_provision(request, appliance, provider, se
         provision_request.wait_for_request(method='ui')
         request.addfinalizer(provision_request.remove_request)
         assert result.validate(wait="60s")
+
+
+@pytest.fixture(scope="module")
+def setup_dynamic_dialog(appliance, custom_instance):
+    # Create custom instance with ruby method
+    code = dedent(
+        """
+        $evm.log(:info, "Hello World")
+        """
+    )
+    instance = custom_instance(ruby_code=code)
+
+    # Create dynamic dialog
+    service_dialog = appliance.collections.service_dialogs
+    dialog = "dialog_{}".format(fauxfactory.gen_alphanumeric())
+    ele_name = "ele_{}".format(fauxfactory.gen_alphanumeric())
+
+    element_data = {
+        "element_information": {
+            "ele_label": "ele_{}".format(fauxfactory.gen_alphanumeric()),
+            "ele_name": ele_name,
+            "ele_desc": fauxfactory.gen_alphanumeric(),
+            "dynamic_chkbox": True,
+            "choose_type": "Text Box",
+        },
+        "options": {"entry_point": instance.tree_path},
+    }
+    sd = service_dialog.create(label=dialog, description="my dialog")
```

Comment:
**Optional:**  I would prefer directly using `appliance.collections.service_dialogs.create()` instead of defining a separate variable for `appliance.collections.service_dialogs`.

Reply:
Since this is a part of the setup, thoughts on creating the service dialog ViaREST? Will make things faster.

Reply:
- Actually this test case demands that **we should select automate instance/method tree via UI**
- I am automating this test case to check the issue of selecting automate tree via UI in service dialog. Aslo I have extended this test case by attaching this service dialog to generic catalog item and checked whether dialog executes attached automate method